src/tools/wix/BinderFileManager.cs

```diff
@@ -735,7 +777,60 @@ private static bool CheckFileExists(string path)
             }
         }
 
+        /// <summary>
+        /// Private method to retry a file action that may block because
+        /// another process is working the file.  Retries on
+        /// an <see cref="IOException" />.
+        /// </summary>
+        /// <param name="path">File path to watch.</param>
+        /// <typeparam name="T">Return type of the file action.</typeparam>
+        /// <param name="func">File <c>I/O</c> Delegate to retry.</param>
+        /// <returns>
+        /// Returns the result of the delegate on success.
+        /// </returns>
+        private T RetryFileAction<T>(string path, Func<T> func)
+        {
+            var autoResetEvent = new AutoResetEvent(false);
+            int retries = 3;
+
+            while (true)
+            {
+                try
+                {
+                    return func();
+                }
+                catch (IOException)
+                {
+                    var fsw = new FileSystemWatcher(Path.GetDirectoryName(path)) { EnableRaisingEvents = true };
```

Comment:
This seems pretty heavyweight. Plus it ends up waiting forever instead of timing out, if there's no event on the file. What if it's a file that's causing the problem but we lock on the directory? Recommend omitting.